deploy/olm-catalog/portworx/core_v1alpha1_storagecluster_crd.yaml

```diff
@@ -28,99 +44,291 @@ spec:
+                    information to authenticate with the KVDB. It could have the username/password
+                    for basic auth, certificate information or an ACL token.
             storage:
+              type: object
+              description: Details of the storage used by the storage driver.
               properties:
                 useAll:
                   type: boolean
+                  description: Use all available, unformatted, unpartioned devices. This will be
+                    ignored if spec.storage.devices is not empty.
                 useAllWithPartitions:
                   type: boolean
+                  description: Use all available unformatted devices. This will be
+                    ignored if spec.storage.devices is not empty.
                 forceUseDisks:
                   type: boolean
-                devices: {}
+                  description: Flag indicating to use the devices even if there is file system present
+                    on it. Note that the devices may be wiped before using.
+                devices:
+                  type: array
+                  description: List of devices to be used by the storage driver.
+                  items:
+                    type: string
                 journalDevice:
                   type: string
+                  description: Device used for journaling.
                 systemMetadataDevice:
                   type: string
-                dataStorageType:
-                  type: string
-                raidLevel:
-                  type: string
+                  description: Device that will be used to store system metadata by the driver.
             cloudStorage:
+              type: object
+              description: Details of storage used in cloud environment.
               properties:
                 maxStorageNodes:
                   type: integer
                   format: int32
                   minimum: 0
+                  description: Maximum nodes that will have storage in the cluster.
                 maxStorageNodesPerZone:
                   type: integer
                   format: int32
                   minimum: 0
-                deviceSpecs: {}
+                  description: Maximum nodes in every zone that will have storage in the cluster.
+                deviceSpecs:
+                  type: array
+                  description: List of storage device specs. A cloud storage device will be created
+                    for every spec in the list. The specs will be applied to all nodes in the cluster
+                    up to spec.cloudStorage.maxStorageNodes or spec.cloudStorage.maxStorageNodesPerZone.
+                  items:
+                    type: string
+                capacitySpecs:
+                  type: array
+                  description: List of cluster wide storage types and their capacities. A single
```

Comment:
nit: may be apply the suggestions from StorageCluster to the status object here as well.